File: frontend/src/types/offer.ts
import type { MusicBranch, OfferStatus } from './enums';

// Sahnem.Business/DTOs/Offer/OfferResponseDto.cs — musicianId, Offer.MusicianId
// (Musician'ın AppUserId'si) alanına karşılık gelir, MusicianProfile.Id'ye DEĞİL.
export interface Offer {
  id: number;
  musicianId: number;
  musicianName: string;
  musicianBranch?: MusicBranch;
  advertId: number;
  advertTitle: string;
  message: string;
  proposedPrice: number;
  offerStatus: OfferStatus;
  createdDate: string;
}

export interface OfferCreateInput {
  advertId: number;
  message: string;
  proposedPrice: number;
}

File: backend/Sahnem.Core/Entities/Offer.cs
using Sahnem.Core.Enums;

namespace Sahnem.Core.Entities
{
    
    public class Offer : BaseEntity
    {
        public int MusicianId{get; set;}
        public int AdvertId{get; set;}
        public string Message{get; set;}
        public decimal ProposedPrice{get; set;}
        public OfferStatus OfferStatus{get; set;} = OfferStatus.Pending;

        public virtual Advert Advert {get; set;}
        public virtual AppUser Musician {get; set;}


    }
}